### PR TITLE
update plan description and add put items

### LINKED ISSUE #92: Plans need to be updated
We have to check the whole Plan description,

an issue in the Create example: `status` field doesn't exist anymore.

### PR DESCRIPTION
**Pull request by GabrielBlondeau-Kirby**
fix #92 

----

Add `PUT /plan/<plan_id>/items` and updates some json example

### CODE REVIEW

**Comment by GabrielBlondeau-Kirby** [comment]
Gonna go on it on Next Friday and see if it's good or if I have to update it.

File: source/includes/_plans.md [hunk 1]
```diff
@@ -115,7 +116,18 @@ Argument | Type | Description
 **interval_unit** | `string` | The frequency that the Subscription acts upon. <br> interval_unit choices: `DAY`, `WEEK`, `MONTH`, `MONTH_END`, `ANNUAL`
 **billing_interval** | `string`| Defines billing frequency. <br> Choices: `WEEK`, `MONTH`, `MONTH_END`
 **currency** | `string`| Three letter ISO currency code as defined by ISO 4217.7
-
+name | `string` | The name of the plan.
+note | `string` | A short description of the plan.
+interval_unit | `string` | The frequency that the subscription acts upon. <br> Choices: `DAY`, `WEEK`, `MONTH`, <br> `MONTH_END`, `ANNUAL`
+interval_count | `integer`| Total number of interval_units.
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: The total number of interval_units the subscription runs before it stops.

File: source/includes/_plans.md [hunk 2]
```diff
@@ -115,7 +116,18 @@ Argument | Type | Description
...
+billing_interval | `string`| Defines billing frequency. <br> Choices: `WEEK`, `MONTH`, `MONTH_END`
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: Defines the payment frequency

File: source/includes/_plans.md [hunk 3]
```diff
@@ -115,7 +116,18 @@ Argument | Type | Description
...
+static | `boolean`| Defines if plan is allowed to be changed.
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: Sets if the params in the params in the plan are changeable after its creation

File: source/includes/_plans.md [hunk 4]
```diff
@@ -115,7 +116,18 @@ Argument | Type | Description
...
+items | `array` | List of items associated with Plan. See description below.
+currency | `string` | Three letter ISO currency code as defined by ISO 4217.
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: The currency of the payment. ISO 4217-standard.

File: source/includes/_plans.md [hunk 5]
```diff
@@ -115,7 +116,18 @@ Argument | Type | Description
...
+total_amount | `integer` | The amount to be charged on the interval specified. If missing, this will be calculated as the sum of the `items`.
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: The amount in the `currency` that will be charged at the end of the billing interval. If missing, the `total_amount` will be calculated as the sum of the `items`.

File: source/includes/_plans.md [hunk 6]
```diff
@@ -115,7 +116,18 @@ Argument | Type | Description
...
+initial_fail_amount_action | `string` | Decides what happens if a payment fails in the [`Subscription`](#subscriptions). <br> Choices: `CANCEL`, `CONTINUE`
+max_fail_attempts | `integer`| If `initial_fail_amount_action` is `CONTINUE`, this is the number of interval_units that is allowed to fail before the [`Subscription`](#subscriptions) stops.
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: If `initial_fail_amount_action` is `CONTINUE`, this is the maximum number of failed payments in a row that is allowed before the [`Subscription`](#subscriptions) stops. If `initial_fail_amount_action` is set to `CANCEL`, the subscription stops if one payment fails.

File: source/includes/_plans.md [hunk 7]
```diff
@@ -115,7 +116,18 @@ Argument | Type | Description
...
+setup_fee | `integer` | If you need to charge a subscription fee to the customer. Can't be less than `0`. _Default `0`_
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: A setup fee that will be paid at the beginning of the first payment cycle.

File: source/includes/_plans.md [hunk 8]
```diff
@@ -28,6 +28,7 @@ currency | `string` | Three letter ISO currency code as defined by ISO 4217.
 total_amount | `integer` | The amount to be charged on the interval specified. If missing, this will be calculated as the sum of the `items`.
 initial_fail_amount_action | `string` | Decides what happens if a payment fails in the [`Subscription`](#subscriptions). <br> Choices: `CANCEL`, `CONTINUE`
 max_fail_attempts | `integer`| If `initial_fail_amount_action` is `CONTINUE`, this is the number of interval_units that is allowed to fail before the [`Subscription`](#subscriptions) stops.
+setup_fee | `integer` | If you need to charge a subscription fee to the customer. Can't be less than `0`. _Default `0`_
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: Could you also add the field `prepay`?

It's boolean and defines if the subscription payment is charged at the beginning or end of the billing cycle.

File: source/includes/_plans.md [hunk 9]
```diff
@@ -278,7 +288,60 @@ initial_fail_amount_action |`string`| Indicates what happens when payment for th
 max_fail_attempts |`number`| How many times a payment can fail in the subscription, but still continue.
 discount |`number`| Between 0.0 an 1.0, where 1.0 is 100% discount on the total_price.
 total_amount |`number`| The total cost of the subscription payment. If set, the total_amount is not based on the sum of products.
+setup_fee | `integer` |
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: Description here?

File: source/includes/_plans.md [hunk 10]
```diff
@@ -154,16 +168,17 @@ Content-Type: application/json
...
      "billing_interval": "MONTH",
      "currency": "NOK", 
      "name": "Golden Plan",
      "interval_unit": "WEEK",
      "static": false,
      "total_amount": 500.0,
-     "company": {
-        "$oid": "57ee9c71d76d431f8511142f"}
+     "company": {"$oid": "57ee9c71d76d431f8511142f"},
+    "setup_fee": true
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: setup fee is a `float`

File: source/includes/_plans.md [hunk 11]
```diff
@@ -403,15 +454,15 @@ Content-Type: application/json
...
          "product": {"$oid": "5931697ed57ba271c0c7de65"},
          "quantity": 2}],
-    "status": "CREATED",
     "currency": "NOK",
     "note": "This is a note regarding the Plan",
     "deleted": false,
     "company": {
         "$oid": "57ee9c71d76d431f8511142f"},
     "static": true,
+    "setup_fee": true,
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: setup_fee is a `float`

File: source/includes/_plans.md [hunk 12]
```diff
@@ -449,6 +500,7 @@ Host: api.kvass.ai
     "note": "This is a note regarding the Plan",
     "interval_count": false,
     "total_amount": false,
+    "setup_fee": true,
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: `float`

File: source/includes/_plans.md [hunk 13]
```diff
@@ -481,13 +531,12 @@ Content-Type: application/json
...
          "quantity": 2}],
-    "status": "CREATED",
     "currency": "NOK",
     "note": "This is a note regarding the Plan",
     "deleted": false,
-    "company": {
-        "$oid": "57ee9c71d76d431f8511142f"},
+    "company": {"$oid": "57ee9c71d76d431f8511142f"},
     "static": true,
+    "setup_fee": true,
```
Comment (KristopherJordan) [inline source/includes/_plans.md]: `float`

**Pull request merged**